src/commands/deploy/setPreDeployTaskForDotnet.ts

```diff
@@ -0,0 +1,160 @@
+
+            const existingTasks: tasks.ITask[] = tasks.getTasks(context.workspace);
+
+            // do not overwrite any dotnet commands the user may have
+            let dotnetTasks: tasks.ITask[] = generateDotnetTasks(subfolder);
+            dotnetTasks = dotnetTasks.filter(t1 => !existingTasks.find(t2 => {
+                return t1.label === t2.label;
+            }));
+
+            const newTasks: tasks.ITask[] = existingTasks.concat(dotnetTasks);
+
+            tasks.updateTasks(context.workspace, newTasks);
+        }
+
+    }
+}
+
+async function tryGetCsprojFile(context: IDeployWizardContext, projectPath: string): Promise<string | undefined> {
+    let projectFiles: string[] = await checkFolderForCsproj(projectPath);
+    // it's a common pattern to have the .csproj file in a subfolder so check one level deeper
+    if (projectFiles.length === 0) {
+        const subfolders: string[] = await fse.readdir(projectPath);
+        await Promise.all(subfolders.map(async folder => {
+            const filePath: string = path.join(projectPath, folder);
+            // check its existence as this will check .vscode even if the project doesn't contain that folder
+            if (fse.existsSync(filePath) && (await fse.stat(filePath)).isDirectory()) {
+                projectFiles = projectFiles.concat(await checkFolderForCsproj(filePath));
+            }
+        }));
+    }
+
+    context.telemetry.properties.numOfCsprojFiles = projectFiles.length.toString();
+
+    // if multiple csprojs were found, ignore them
+    return projectFiles.length === 1 ? projectFiles[0] : undefined;
+
+    async function checkFolderForCsproj(filePath: string): Promise<string[]> {
+        const files: string[] = fse.readdirSync(filePath);
+        const filePaths: string[] = files.map((f: string) => {
+            return path.join(filePath, f);
+        });
+
+        return filePaths.filter((f: string) => /\.csproj$/i.test(f));
+    }
+}
+
+async function tryGetTargetFramework(projFilePath: string): Promise<string | undefined> {
+    const projContents: string = (await fse.readFile(projFilePath)).toString();
+    const matches: RegExpMatchArray | null = projContents.match(/<TargetFramework>(.*)<\/TargetFramework>/);
+    return matches === null ? undefined : matches[1];
+}
+
+function generateDotnetTasks(subfolder: string): TaskDefinition[] {
+    // always use posix for debug config
+    // tslint:disable-next-line: no-unsafe-any no-invalid-template-strings
+    const cwd: string = path.posix.join('${workspaceFolder}', subfolder);
+
+    const buildTask: TaskDefinition = {
+        label: "build",
+        command: "dotnet",
+        type: "process",
+        args: [
+            "build",
+            cwd,
+            "/property:GenerateFullPaths=true",
+            "/consoleloggerparameters:NoSummary"
+        ],
+        problemMatcher: "$msCompile"
+    };
+
+    const publishTask: TaskDefinition = {
+        label: "publish",
```

Comment:
Pending discussion on a few things:
1. Should this depend on a clean task
1. Should this be "Release" instead of "Debug"
1. Should we use the "publish" label or create our own name to differentiate from C# extension

Reply:
> [4:38 PM] Ahmed Metwally
>     1- We should call clean at first.
> 2- We can default to debug to be like everyone else.
> 3- We can reuse the C# extension publish task or create one if it doesn't exist.